### PR TITLE
correct number of samples in loss calculation

### PR DESCRIPTION
**Pull request by pzhanggit**
Total number of samples (n_tot) in the dataset was used in calculating losses. This was fine in single rank runs.
But when multiple ranks were involved in DDP, the loss calculated would be less that the actual loss, since actual samples were split to each rank and the number of samples on each rank should be smaller (n_tot/n_ranks), but still the total number (n_tot) were used. 
This (probably) partially explains the large loss difference in multiple GPUs vs single GPU.

### CODE REVIEW

**Review by allaffa — APPROVED** [review]

**Review by jychoi-hpc — APPROVED** [review]
Make sense to me. It should work with DDP too.

File: hydragnn/train/train_validate_test.py [hunk 1]
```diff
@@ -211,11 +211,11 @@ def train(
 ):
     if profiler is None:
         profiler = Profiler()
-    tasks_error = np.zeros(model.num_heads)
-
-    model.train()
 
     total_error = 0
+    tasks_error = np.zeros(model.num_heads)
+    num_samples_total = 0
```
[the remark this answers is not in the document]
Comment (pzhanggit) [reply hydragnn/train/train_validate_test.py]: good comment, done.

**Review by streeve — APPROVED** [review]
